src/main/java/org/bankofspring/dao/jdbc/BankOfSpringDAOImpl.java

```diff
@@ -0,0 +1,183 @@
+package org.bankofspring.dao.jdbc;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.bankofspring.dao.BankOfSpringDAO;
+import org.bankofspring.model.Account;
+import org.bankofspring.model.AccountTransaction;
+import org.bankofspring.model.Customer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.core.simple.SimpleJdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.stereotype.Repository;
+@Repository
+public class BankOfSpringDAOImpl implements BankOfSpringDAO {
+	
+	private final String  CUSTOMER_SELECT = "SELECT c.id, c.name, c.username, u.password from customer c inner join user u on (c.username = u.username)";
+	private final String  CUSTOMERS_SELECT = CUSTOMER_SELECT +  " order by c.id";
+	private final String  SINGLE_CUSTOMER_SELECT =  CUSTOMER_SELECT + "where c.id = ?";
+		private final String  CUSTOMERS_FOR_ACCOUNT_SELECT = "SELECT c.id, c.name, c.username, u.password from customer c "+
+							"inner join user u on (c.username = u.username) inner join customer_account a "
+							+ "on (a.customer_id = c.id) where a.number =?";
+	private final String TXN_ACCOUNT_SELECT = "select id,from_account_number, to_account_number, time, amount from account_transaction where"
+			+ " id = ?";
+	private final String TXN_ACCOUNT_ADD = "insert into account_transaction (from_account_number, to_account_number,  amount, time) values " +
+			"( :from_account_number,:to_account_number,:amount,:time)";
+
+	private final String  ACCOUNT_SELECT =  "SELECT number, description,balance from account where number=?";
+	private final String ACCOUNT_UPDATE = "UPDATE account set balance=? where number = ?";
+	
+	private SimpleJdbcTemplate simpleTemplate;
+	private NamedParameterJdbcTemplate namedTemplate;
+	
+	@Autowired
+	public void setDataSource(DataSource source) {
+		this.simpleTemplate = new SimpleJdbcTemplate(source);
+		this.namedTemplate = new NamedParameterJdbcTemplate(source);
+	}
+	
+	public SimpleJdbcTemplate getSimpleTemplate() {
+		return simpleTemplate;
+	}
+	
+	public NamedParameterJdbcTemplate getNamedTemplate() {
+		return namedTemplate;
+	}
+	
+	@Override
+	public List<Customer> getCustomers() {
+		 return getSimpleTemplate().query(CUSTOMERS_SELECT, new RowMapper<Customer>() {
+
+			@Override
+			public Customer mapRow(ResultSet res, int rowNum)
+					throws SQLException {
+				String id = res.getString(1);
```

Comment:
Not a big deal for this, but it is sometimes better to use the actual names, so it doesn't break if someone selects an extra column in the query.